fiksi/src/lib.rs

```diff
@@ -438,6 +449,8 @@ impl System {
     /// element the handle corresponds to is considered free, regardless of whether the circle
     /// itself is in `solve_set`.
     pub fn solve(&mut self, solve_set: Option<&SolveSetHandle>, opts: SolvingOptions) {
+        let mut rng = Rng::from_seed(42);
```

Comment:
Will we ever care about setting the seed?

Reply:
I'm not certain. Practically it may never be necessary, but theoretically it's possible that a system doesn't solve with one seed but does solve with another. That's more likely to happen when getting stuck in an attracting basin that's a local optimum, rather than due to saddle ridges. With good decomposition getting stuck in local optima is less likely in general.

A possibility when a subsystem wasn't solved the first time, but heuristically we think it's solvable, is is to retry solving it with different or more perturbation.

Reply:
(Currently the perturbation in this PR is really only designed to deal with saddle ridges / singular states.)